aten/src/ATen/core/ivalue.cpp

```diff
@@ -630,6 +631,29 @@ getClassConverter() {
   return classConverter;
 }
 
+void ivalue::Future::addCallback(
+    std::function<void(void)> callback,
+    bool propagateTLSState) {
```

Comment:
Do we need this flag? Why don't we always propagate the TLS state? (similar to what we do in at::launch()). If this is optional, we'll mostly run in to issues time and again where we encounter a bug and realize we need to set this to true.

If we would end up breaking some existing code with such a change, maybe make the default value of the flag true and fix the handful of places by setting this to false.

Reply:
ivalue::Future seems to be a fairly generic-intended api.

It feels like a strong position to say that we unconditionally want it to propagate TLS state - this will get expensive over time, and I can imagine low-level operations where this might be be particularly appropriate.

Supporting it here with a wrapper or param seems reasonable, to avoid boilerplate.
But forcing it in every case may be excessively heavyweight.

Reply:
I think propagating it by default might warrant some more discussion. Maybe for this PR we can add @jjlilley 's suggestion of the wrapper function, and use in RPC-related callsites. If we do decide to unconditionally propagate it should be an easy change to just have `addCallback` call into this helper, and remove the helper from all callsites.